### PR TITLE
feat: Added DeviceInfo.supportedABIs()

### PR DESCRIPTION
**Pull request by hemantasapkota**
## Description

Fixed issue #578 

Added `supportedABIs()` which returns a list of supported device cpu types

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ✅     |
| Windows |    ❌     |

## Checklist

* [x] I have tested this on a device/simulator for each compatible OS
* [x] I added the documentation in `README.md`.
* [x] I mentioned this change in `CHANGELOG.md`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -253,6 +253,7 @@ import DeviceInfo from 'react-native-device-info';
 | [getDeviceType()](#getDeviceType)                 | `string`            |  ✅  |   ✅    |   ❌    | ?      |
 | [isAutoDateAndTime()](#isAutoDateAndTime)         | `boolean`           |  ❌  |   ✅    |   ❌    | 0.29.0 |
 | [isAutoTimeZone()](#isAutoTimeZone)               | `boolean`           |  ❌  |   ✅    |   ❌    | 0.29.0 |
+| [supportedABIs()](#supportedABIs)                 | `string[]`          |  ✅  |   ✅    |   ❌    | 1.0.1  |
```
Comment (machour) [inline README.md]: Will be 1.1.0. Use that to update the Changelog.md file too 🙏

Comment (hemantasapkota) [reply README.md]: done. 🙏

Comment (hemantasapkota) [reply README.md]: sorry, has been fixed now.

File: example/App.js [hunk 2]
```diff
@@ -96,12 +97,12 @@ export default class App extends Component<Props> {
 
   render() {
     return (
-      <View style={styles.container}>
+      <SafeAreaView style={styles.container}>
```
Comment (mikehardy) [inline example/App.js]: SafeAreaView was new to me! I'm still new at the react-native level of things. This is a nice touch, thanks

File: android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java [hunk 3]
```diff
@@ -395,7 +395,9 @@ public String getInstallReferrer() {
     actMgr.getMemoryInfo(memInfo);
     constants.put("totalMemory", memInfo.totalMem);
     constants.put("deviceType", deviceType.getValue());
-
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
```
Comment (mikehardy) [inline android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java]: I like the API-gate here for the array version of Build.SUPPORTED_ABIS, but this means that in lower APIs (down to API16, React-Native's lower bound) we will not have an entry in constants which will force a logic branch to any consumers of the API, and they won't get any info. I think on the earlier devices you can make an empty array then use Build.CPU_ABI like so https://stackoverflow.com/a/51775541 but add it as [0] in the array

I intend to use this feature almost immediately in my app, down to ABI16 to fetch arch-specific flavors in rn-update-apk so this is important to me, and I'll also serve as a test for it pretty quickly :-). Thanks!

File: example/App.js (hunk 2 above)
Comment (hemantasapkota) [reply example/App.js]: glad you like it :)

File: android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java (hunk 3 above)
Comment (hemantasapkota) [reply android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java]: fixed.

Comment (hemantasapkota) [reply android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java]: I tested it on a the emulator with API level 16. Fallback seems to work. I'm hoping it'll work as expected on the device too.

**Comment by mikehardy** [comment]
@hemantasapkota this looks great, I'm excited to have this feature in the module. @machour I looked closely and I can't see anything else I'd change in here? I'm not perfect those of course so I'll leave the merge for you in case you want to take another look

**Comment by hemantasapkota** [comment]
@mikehardy thanks for the review. This has been a good experience. looking forward to @machour 's review.

**Comment by mikehardy** [comment]
Whether I got that merge correct or not :pray: - it is my first merge into this repo. Thanks @hemantasapkota for the work and @machour for the help

**Comment by machour** [comment]
👌

react-native-device-info 1.1.0 published 🎉

**Comment by mikehardy** [comment]
I can see in the commit log the way I did it duplicated the message (the title was the same as the commit message) which isn't as tidy as your previous ones. I'll make sure the title is complete/correct next time, but leave the message empty and they'll match up. Not an A+ yet ;-)

**Pull request merged**